component/src/shielded_pool/note_manager.rs

```diff
@@ -66,7 +66,7 @@ pub trait NoteManager: StateWrite {
         let encrypted_note = note.encrypt(&esk);
 
         // Now record the note and update the total supply:
-        self.update_token_supply(&value.asset_id, i64::from(value.amount))
+        self.update_token_supply(&value.asset_id, value.amount.value() as i64)
```

Comment:
Followup: not in this PR, but we have some cleanup to do around the token supply tracking -- it should use `Amount`s, but also, we should check that whenever we convert tokens from one type to another, we correctly update the supply. This will be tricky for cases like swaps, unbonding, etc., where the chain might not be able to perfectly track that information, and we need to do some design thinking about what the implications of that are...

Reply:
Is there already a related issue for this? Sounds like it could be several tickets

Reply:
No, we should create one as a followup, and the first subtask should be to start with writing down every place where we create or destroy tokens, so we can think through those issues.

Reply:
#1823